Runtimes/Supplemental/Synchronization/CMakeLists.txt

```diff
@@ -141,6 +141,31 @@ target_link_libraries(swiftSynchronization PRIVATE
   $<$<PLATFORM_ID:Darwin>:swiftDarwin>
   $<$<PLATFORM_ID:Windows>:ClangModules>)
 
+if(WIN32 AND CMAKE_SYSTEM_PROCESSOR STREQUAL i686)
+  # FIXME `-whole-module-optimization` should not be needed. However, without
+  # this, we will not properly optimize. On Windows i686 in particular, we
+  # observe the formation of `__atomic_load` calls rather than the expected
+  # native sequence.
+  #
+  # WMO requires the early-swift-driver to be usable, which is not yet ready on
```

Comment:
We should probably have an issue saying to remove this once Windows has the necessary support.